hydrolib/core/utils.py

```diff
@@ -300,3 +301,34 @@ def _from_windows_to_posix_path(cls, windows_path: Path) -> str:
         posix_path = posix_root + "/".join(parts)
 
         return posix_path
+
+
+class FileChecksumCalculator:
+    """
+    FileChecksumCalculator calculator used to calculate the checksum of a file.
+    """
+
+    @staticmethod
+    def calculate_checksum(filepath: Path) -> Optional[str]:
+        """Calculate the checksum of the file from the given filepath.
+
+        Args:
+            filepath (Path): The filepath to the file for which the checksum will be calculated.
+
+        Returns:
+            [Optional[str]]:
+                The checksum of the file.
+                When the filepath doesn't exist or the filepath isn't a file, None.
+        """
+        if not filepath.exists() or not filepath.is_file():
```

Comment:
Is there a reason you chose to return `None` instead of raising an error?

Reply:
it has to do with the weird files it tries to access on each run, which we also encountered when we were pair programming on this.